independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/jaxrs/SseEventSinkImpl.java

```diff
@@ -75,10 +75,13 @@ public void accept(Throwable throwable) {
                     // I don't think we should be firing the exception on the broadcaster here
                 }
             });
-            //            response.closeHandler(v -> {
-            //                // FIXME: notify of client closing
-            //                System.err.println("Server connection closed");
-            //            });
+            response.addCloseHandler(() -> {
```

Comment:
Hi!

I checked out your branch and was testing the solution but I cannot understand why the close handler is only added when `!response.headWritten()`. Is this on purpose? For some reason, on my application this condition is always false and because of that the Close event firing never happens.

Reply:
Hi @rjvq85, thanks for taking a look! 
No, this was not placed there with a specific purpose, I just added the closeHandler where the previous comment was, but it makes sense to move it out of the if statement, to always add it to the response. 

For me the condition `!response.headWritten()` was always true when testing, so it always execute the close handler, and I cannot reproduce a case where it gets evaluated to false... I used the reproducer provided by #23997, with the `sse-server-reactive` module and it worked fine, the issues on the reactive server seemed to be resolved.

I tested again after moving the changes out of the if statement, and I don't see any unexpected behaviour, so i guess it's fine like that?